xbmc/platform/darwin/ios-common/DarwinEmbedNowPlayingInfoManager.mm

```diff
@@ -0,0 +1,132 @@
+/*
+ *  Copyright (C) 2010-2019 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#import "DarwinEmbedNowPlayingInfoManager.h"
+
+#import "platform/darwin/ios/XBMCController.h"
+
+#import <MediaPlayer/MediaPlayer.h>
+
+@implementation DarwinEmbedNowPlayingInfoManager
+
+@synthesize nowPlayingInfo = m_nowPlayingInfo;
+@synthesize playbackState;
+
+#pragma mark - Now Playing routines
+
+- (void)setNowPlayingInfo:(NSDictionary*)info
+{
+  m_nowPlayingInfo = info;
```

Comment:
your implementation is incorrect :) correct way is something like
```
if (![m_nowPlayingInfo isEqualToDictionary:info])
{
  m_nowPlayingInfo = [info copy];
  [MPNowPlayingInfoCenter defaultCenter].nowPlayingInfo = m_nowPlayingInfo;
}
```

but to avoid implementing default behavior manually better use KVO :) I can help you with this.